apps/framework-cli/src/framework/python/wrappers/scripts/python_worker_wrapper/activity.py

```diff
@@ -51,11 +53,28 @@ async def dynamic_activity(execution_input: ScriptExecutionInput) -> Any:
             if not task_func:
                 raise ValueError("No @task function found in script.")
             
-            # Execute
-            if execution_input.input_data:
-                return task_func(**execution_input.input_data)
+            # Execute with data from previous step if available
+            if execution_input.input_data and "data" in execution_input.input_data:
```

Comment:
I think we need to check if this changes the way initial input data works through the CLI & API

`moose-cli workflow run <name> --input '<json-data>'`

Reply:
Nice catch. updated PR